qcodes/instrument_drivers/rohde_schwarz/ZNB.py

```diff
@@ -1,16 +1,143 @@
 import logging
+import numpy as np
 from functools import partial
 from typing import Optional, Any, Tuple
 
 from qcodes import VisaInstrument, Instrument
 from qcodes import ChannelList, InstrumentChannel
 from qcodes.utils import validators as vals
-import numpy as np
-from qcodes.instrument.parameter import MultiParameter, ArrayParameter, ParamRawDataType
+from qcodes.instrument.parameter import MultiParameter, ArrayParameter, \
+    ParamRawDataType
+from qcodes.utils.deprecate import deprecate
+
 
 log = logging.getLogger(__name__)
 
 
+class FixedFrequencyTraceIQ(MultiParameter):
+    """
+    Sweep that returns the real (I) and imaginary (Q) parts of the VNA response.
+    Requires the use of the sweep type to be set to continuous wave mode.
+    See (https://www.rohde-schwarz.com/webhelp/ZNB_ZNBT_HTML_UserManual_en
+    /ZNB_ZNBT_HTML_UserManual_en.htm) under GUI reference -> sweep softtool
+    -> sweep type tab -> CW mode
+    """
+
+    def __init__(self, name: str, instrument: "ZNBChannel",
+                 npts: int, bandwidth: int, channel: int,
+                 check_cw_sweep_first: bool = True) -> None:
+        super().__init__(name, names=("", ""), shapes=((), ()))
+        self._instrument = instrument
```

Comment:
Please pass the instrument to the super class and let that asign to `self._instrument` rather than repeating the work here. 

Multiple of these below in other parameters too